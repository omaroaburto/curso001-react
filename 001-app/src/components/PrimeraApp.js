import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ({saludo, mensaje}) => {
    return (
        <>
            <h1>{saludo}</h1>
            <span>{mensaje}</span>   
        </>
    )
}

PrimeraApp.propTypes={
    saludo : PropTypes.string.isRequired,
    mensaje: PropTypes.string.isRequired
};

PrimeraApp.defaultProps ={
    saludo: 'Hola gente',
    mensaje: 'Mi nombre es omaro'
}

export default PrimeraApp;