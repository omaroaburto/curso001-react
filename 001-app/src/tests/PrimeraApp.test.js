import React from 'react';  
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import PrimeraApp from '../components/PrimeraApp';
describe('Pruebas del archivo PrimeraApp.test.js',()=>{
    test('Debe mostrar el mensaje Hola soy omaro',()=>{
        const saludo =  "Hola soy omaro";
        const mensaje =  "Soy ingeniero inform√°tico";
        //const {getByText} =  render(<PrimeraApp saludo = {saludo} />)
        //expect(getByText(saludo)).toBeInTheDocument();

        const wrapper =  shallow(<PrimeraApp saludo = {saludo} mensaje = {mensaje}/>);
        expect(wrapper).toMatchSnapshot() 
    }); 
});