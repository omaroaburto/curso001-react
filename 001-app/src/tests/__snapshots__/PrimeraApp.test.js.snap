// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del archivo PrimeraApp.test.js Debe mostrar el mensaje Hola soy omaro 1`] = `
<Fragment>
  <h1>
    Hola soy omaro
  </h1>
  <span>
    Soy ingeniero inform√°tico
  </span>
</Fragment>
`;
